### PR TITLE
Reorganize ebird code

### PR DESCRIPTION
**Pull request by bsaul**

### CODE REVIEW

File: R/ebird_import.R [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+  `ML Catalog Numbers`     = col_skip()
+)
+
+#' Load the *most recent* ebird export per county
+#' (i.e. MBBS ebird account)
+#' into an R `data.frame`.
+#' according the `ebird_cols` specification.
+#' Additionally renames the variables.
+#'
+#' NOTE: readr::read_csv gives warnings upon parsing the ebird CSVs.
```
Comment (bsaul) [inline R/ebird_import.R:57]: Create issue regarding this note.

**Comment by bsaul** [comment]
Processing ebird comments will be handled in a later PR.

**Comment by bsaul** [comment]
@IJBG - this one is ready for review too, but #104 should come first.

**Review by IJBG — APPROVED** [review]

File: R/ebird_import.R (hunk 1 above)
Comment (bsaul) [reply R/ebird_import.R:57]: Yeah, for all the csvs. You can see in the first non-header row of (e.g. `MyEBirdData_Chatham_20240730.csv`): there's no comma after the checklist comment, but -- per the header -- there should be one for `ML Catalog Numbers`.

This reminds me though, we need to add a note somewhere to download ebird files directly and **not open with excel**.

File: R/ebird_comments.R [hunk 2]
```diff
@@ -35,7 +34,7 @@ make_comment_extractor <- function(field_pattern, data_pattern, delimiter = ";",
   )
 
   function(comments, ...) {
-    stringr::str_extract_all(comments, pattern = valid_pattern) %>%
+    stringr::str_extract_all(comments, pattern = valid_pattern) |>
```
[the remark this answers is not in the document]
Comment (bsaul) [reply R/ebird_comments.R:37]: `|>` is R's built-in pipe that was added in v4.ish -- after this project got started. It doensn't really matter much which you use, I prefer the built-in now that the language has it. I've been switching over as I go, but I don't think we need to make a special effort to switch.

**Pull request merged**